web/app/common/constants.ts

```diff
@@ -9,35 +9,43 @@ export enum BuildStatus {
 export type FastlaneStatus =
     'failure'|'success'|'ci_problem'|'pending'|'missing_fastfile';
 
-export function fastlaneStatusToEnum(status: FastlaneStatus): BuildStatus {
-  switch (status) {
-    case 'success':
-      return BuildStatus.SUCCESS;
-    case 'failure':
-      return BuildStatus.FAILED;
-    case 'pending':
-      return BuildStatus.PENDING;
-    case 'missing_fastfile':
-      return BuildStatus.MISSING_FASTFILE;
-    case 'ci_problem':
-      return BuildStatus.INTERNAL_ISSUE;
-    default:
-      throw new Error(`Unknown status type ${status}`);
+export function fastlaneStatusToEnum(status?: FastlaneStatus): BuildStatus {
```

Comment:
i think it might be better to have the field undefined in the model itself since logically there is no status.